test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -45,6 +48,36 @@ public void AutoFitMultiTest()
             Assert.Equal(NumberDataViewType.Single, scoredData.Schema[DefaultColumnNames.PredictedLabel].Type);
         }
 
+        [TensorFlowFact]
+        public void AutoFitImageClassificationTrainTest()
+        {
+            var context = new MLContext();
+            var datasetPath = DatasetUtil.GetFlowersDataset();
+            var columnInference = context.Auto().InferColumns(datasetPath, "Label");
+            var textLoader = context.Data.CreateTextLoader(columnInference.TextLoaderOptions);
+            var trainData = context.Data.ShuffleRows(textLoader.Load(datasetPath), seed: 1);
+            var originalColumnNames = trainData.Schema.Select(c => c.Name);
+            TrainTestData trainTestData = context.Data.TrainTestSplit(trainData, testFraction: 0.2, seed: 1);
+            IDataView trainDataset = SplitUtil.DropAllColumnsExcept(context, trainTestData.TrainSet, originalColumnNames);
+            IDataView testDataset = SplitUtil.DropAllColumnsExcept(context, trainTestData.TestSet, originalColumnNames);
+            var result = context.Auto()
+                            .CreateMulticlassClassificationExperiment(0)
+                            .Execute(trainDataset, testDataset, columnInference.ColumnInformation);
+
+            //Known issue, where on Ubuntu there is degradation in accuracy.
+            if (!(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ||
+                (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))))
+            {
+                Assert.Equal(0.889, result.BestRun.ValidationMetrics.MicroAccuracy, 3);
+            }
+            else
+            {
+                Assert.Equal(0.444, result.BestRun.ValidationMetrics.MicroAccuracy, 3);
```

Comment:
What's causing the `ImageClassificationTrainer` (or another component) to fail on Linux? What's the timeline for a fix?

@rustd, @CESARDELATORRE -- If we have to, is shipping local image classification without Linux support ok?

Reply:
First, its not failing on Linux :) we see not so good accuracy on Linux for *some* datasets (mostly smaller ones .. our samples seem to work great) but on Windows and MacOS accuracy is consistent. We haven't investigated why for some smaller test datasets Linux gives not so good accuracy but may be after release we can do that.

---
In reply to: [339356839](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339356839) [](ancestors = 339356839)